### PR TITLE
Python based testing

### PR DESCRIPTION
**Pull request by PeterDuerr**
In order to facilitate adding tests for ecl (unfortunately I am not that fond of writing unit tests in C++ :-)), I wrote a (very simple) Python wrapper based on SWIG. Exploiting Python libraries (pytest, hypothesis, and optionally matplotlib), one can now easily add assert-based tests, property-based tests, as well as plot data during (selected) tests.

To test my work, please checkout my branch and run `cmake -DPythonTests=1 ../EKF && make pytest-quick`. You can also run `make pytest-plots` to see graphs like the attached...

If you agree with this general approach, one could refactor the main classes in the C++ code slightly to make the wrapper more elegant and powerful.

Before I go any further though, please let me know what you think. I would love to hear your feedback. 

Peter

test_accel_z_bias_converges

### CODE REVIEW

**Comment by dagar** [comment]
Awesome! We should talk about getting this into the CI system.

File: EKF/tests/pytest/lint [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+###############################################################################
```
Comment (dagar) [inline EKF/tests/pytest/lint]: No .py file extension?

**Comment by dagar** [comment]
Can you list the package dependencies?

File: EKF/tests/pytest/lint (hunk 1 above)
Comment (PeterDuerr) [inline EKF/tests/pytest/lint]: AFAIK, no .py file extension is quite standard for executable Python scripts (see, e.g., the `pip` command).

**Comment by PeterDuerr** [comment]
@dagar In the latest commit I added a `requirements.txt` file and instead of having the hard-coded list in `CMakeLists.txt` as before, I read it at configuration time. Hope that helps...

**Comment by LorenzMeier** [comment]
We have Travis enabled on this repo - would you be interested to extend the .travis file to run the tests you added?

**Comment by PeterDuerr** [comment]
@LorenzMeier Hi Lorenz, sure, it should be fairly straight-forward to do so... I'll have a look on Monday.

**Comment by LorenzMeier** [comment]
Awesome!

**Pull request merged**